subprojects/dependency-management/src/main/java/org/gradle/internal/locking/NoOpDependencyLockingProvider.java

```diff
@@ -24,18 +24,15 @@
 import org.gradle.api.provider.ListProperty;
 import org.gradle.api.provider.Property;
 
+import javax.annotation.Nullable;
 import java.util.Set;
 
 public class NoOpDependencyLockingProvider implements DependencyLockingProvider {
 
-    private static final NoOpDependencyLockingProvider INSTANCE = new NoOpDependencyLockingProvider();
+    private final DependencyLockingProvider delegate;
 
-    public static DependencyLockingProvider getInstance() {
-        return INSTANCE;
-    }
-
-    private NoOpDependencyLockingProvider() {
-        // Prevent construction
+    public NoOpDependencyLockingProvider(@Nullable DependencyLockingProvider delegate) {
+        this.delegate = delegate;
```

Comment:
This looks like the name no longer matches the behaviour of this class. It seems like the `NoOpDependencyLockingProvider` is now operating as a `DecoratingDependencyLockingProvider`. Can you clarify why this hasn't just become a decorator?

Reply:
Renamed to `NoLockingDependencyLockingProvider` and added class level javadoc to clarify its purpose.

Reply:
Actually ended up reworking this. The `NoOpDependencyLockingProvider` is back to what it was and other places have been cleaned up.
Thanks for making me think twice about it!